qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -119,4 +125,26 @@ def predict(self, X: np.ndarray) -> np.ndarray:  # pylint: disable=invalid-name
     def score(
         self, X: np.ndarray, y: np.ndarray, sample_weight: Optional[np.ndarray] = None
     ) -> float:
+        y = self._encode_labels(y)
         return ClassifierMixin.score(self, X, y, sample_weight)
+
+    def _fit_and_encode_labels(self, y: np.ndarray) -> np.ndarray:
+        """Fits label or one-hot encoder and converts categorical target data."""
+
+        if isinstance(y[0], str):
+            # string data is assumed to be categorical
+            y = y.reshape(-1, 1) if self._one_hot else y  # reshape labels for encoder
```

Comment:
So now I'm confused. Why you added `if self._one_hot else y`?

Reply:
That was that bug I breifly mentioned earlier. `OneHotEncoder` expects input of shape `(n_samples, n_features)` (see [here](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html#sklearn.preprocessing.OneHotEncoder.fit)) while `LabelEncoder` takes in data with shape `(n_samples,)` (see [here](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html#sklearn.preprocessing.LabelEncoder.fit)) so it was just printing a warning whenever someone uses the `LabelEncoder`.

Reply:
Thanks a lot, would you mind to add a comment with this explanation to the code? Just one or two lines should be enough, let's make life easier for future readers. Other than that I don't have anything on my list.

Reply:
Sounds good!  Added in [3f7c632](https://github.com/Qiskit/qiskit-machine-learning/pull/143/commits/3f7c632a6398b004ab3b42042a21a1011ddc9c83)